docs/azure.rst

```diff
@@ -345,6 +345,42 @@ Public images from other registries are still pulled (if requested by a Task) wh
   specified via the :ref:`container <process-container>` directive using the format: ``[server]/[your-organization]/[your-image]:[tag]``.
   Read more about image fully qualified image names in the `Docker documentation <https://docs.docker.com/engine/reference/commandline/pull/#pull-from-a-different-registry>`_.
 
+Active Directory Authentication
+===============================
+
+As of version ``22.11.0-edge``, `Service Principal <https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal>`_ credentials can optionally be used instead of Shared Keys for Azure Batch and Storage accounts. 
```

Comment:
Would not be better to call it "Azure Active Directory service principal" ?

Reply:
Paolo, the plan is to accommodate other identity within this scope. For example, managed identity and user identity is a type of service principals (with slightly different behavior) therefore it makes sense to use the naming convention to highlight this. 

Therefore if we continue with current design, these future options would be like 

```groovy
azure {
  // common attributes
  tenantId = '<YOUR TENANT ID>'
  subscriptionId = '<YOUR SUBSCRIPTION ID>'

  activeDirectory {
  
      // If vanilla service principal is used 
      servicePrincipalId = '<YOUR SERVICE PRINCIPAL CLIENT ID>'
      servicePrincipalSecret = '<YOUR SERVICE PRINCIPAL CLIENT SECRET>'
  
      // If managed-identity is used 
      useManagedIdentity = true
     
      // If user-identity is used 
      userAssignedIdentityClientID = <YOUR USER ASSIGNED IDENTITY CLIENT IDENTIFIER>
  
      
    }
}

```

However, if we are okay with another level of nesting then we can do something like 

```groovy
azure {

  // common attributes
  tenantId = '<YOUR TENANT ID>'
  subscriptionId = '<YOUR SUBSCRIPTION ID>'

  activeDirectory { 

      // If vanilla service principal is used 
      servicePrincipal {
            id = '<YOUR SERVICE PRINCIPAL CLIENT ID>'
            secret = '<YOUR SERVICE PRINCIPAL CLIENT SECRET>'
           }
  
      // If managed identity is used 
      managedIdentity { 
           enabled = true
          }
   
      // If user identity is used 
      userIdentity { 
           clientId =  <YOUR USER ASSIGNED IDENTITY CLIENT IDENTIFIER>
         }
  }

}
```

Happy to accommodate whatever seems better and more intuitive.

Also, perhaps we should add the note for `<EXPERIMENTAL SUPPORT>` till we stabilize the entire scope and achieve the overall goal of credential-less auth within Azure?

Reference: https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#managed-identity-types

Reply:
Ok, I think it's fine 👍